plugins/services/src/js/data/ServicePlansModel.ts

```diff
@@ -0,0 +1,123 @@
+import { Observable } from "rxjs";
+import { RequestResponse } from "@dcos/http-service";
+import {
+  fetchPlans as fetchServicePlans,
+  fetchPlanDetails as fetchServicePlanDetail,
+  ServicePlanResponse
+} from "#PLUGINS/services/src/js/data/ServicePlansClient";
+import { makeExecutableSchema } from "graphql-tools";
+
+import Config from "#SRC/js/config/Config";
+import { ServicePlanStatusSchema } from "#PLUGINS/services/src/js/structs/ServicePlanStatus";
+import { ServicePlanStepSchema } from "#PLUGINS/services/src/js/structs/ServicePlanStep";
+import { ServicePlanPhaseSchema } from "#PLUGINS/services/src/js/structs/ServicePlanPhase";
+import {
+  ServicePlan,
+  ServicePlanDetailResolver,
+  ServicePlanSchema
+} from "#PLUGINS/services/src/js/structs/ServicePlan";
+
+export interface Query {
+  servicePlans: string[] | null;
+  servicePlan: ServicePlan | null;
+}
+
+export interface ResolverArgs {
+  fetchServicePlans: (
+    serviceName: string
+  ) => Observable<RequestResponse<string[]>>;
+  fetchServicePlanDetail: (
+    serviceName: string,
+    planName: string
+  ) => Observable<RequestResponse<ServicePlanResponse>>;
+  pollingInterval: number;
+}
+
+export interface GeneralArgs {
+  [key: string]: any;
+}
+
+export interface ServicePlansQueryArgs {
+  serviceName: string;
+}
+
+export interface ServicePlanDetailQueryArgs {
+  serviceName: string;
+  planName: string;
+}
+
+export const typeDefs = `
+${ServicePlanStatusSchema}
```

Comment:
I wonder if it would make sense to move these into separate files and use the 

```graphql
extend ServicePlan {
   steps: ...
}
```

Syntax together with passing them into `makeExecutableSchma` as arrays. 

This would allow us to have a nice place for the field resolvers if there were some to come, plus we can unit test these. Might be overengineered, I don't know 🙈